modules/common/karpenter/main.tf

```diff
@@ -0,0 +1,171 @@
+locals {
+  node_subnet_ids = slice(var.subnet_ids, 0, var.az_count)
+  node_classes = {
+    cpu = {
+      name   = "cpu"
+      ami_id = data.aws_ami.default_eks_node.id
+    }
+    gpu = {
+      name   = "gpu"
+      ami_id = data.aws_ami.gpu_eks_node.id
+    }
+  }
+
+  cpu_instance_requirements = {
+    instance-family = {
+      key       = "karpenter.k8s.aws/instance-family"
+      operator  = "In"
+      values    = ["m5", "m6i", "m6a", "m7i", "m7a", "c5", "c6i", "c6a", "c7i", "c7a", "c7i-flex", "r5", "r6i", "r6a", "r7i", "r7a"]
+      minValues = 16
+    }
+  }
+  gpu_instance_requirements = {
+    instance-family = {
+      key       = "karpenter.k8s.aws/instance-family"
+      operator  = "In"
+      values    = ["g4dn", "g5", "g6"]
```

Comment:
Since we are not tying to launch spot instances, I think we should keep to just the g4dn class of node, we know that they are available and work with out workload.